components/form/TimesheetForm.tsx

```diff
@@ -55,81 +55,151 @@ export function CreateTimesheetForm({
         }
     };
 
-    const onSubmit = async (e: React.MouseEvent) => {
-        e.preventDefault();
-        try {
-            const timesheet = validateTimesheetFields(timesheetFields, projectId);
-            const newTimesheet = await postTimesheet(timesheet, errorHandler);
-            afterSubmit && afterSubmit(newTimesheet);
-        } catch (error) {
-            errorHandler(error as Error);
-        }
-    };
-
     const invalidAllocation =
         (timesheetFields.allocation && (timesheetFields.allocation < 0 || timesheetFields.allocation > 100)) || false;
 
+    const abortSubmission = onCancel ? onCancel : () => undefined;
+
+    const timesheetStatuses = { active: 'ACTIVE', inactive: 'INACTIVE', archived: 'ARCHIVED' };
```

Comment:
I'm not sure what's going on here, but I don't think you need these, since the openapi generator can infer these from the docs

https://github.com/three-consulting/epoc-frontend/blob/main/lib/types/api.ts#L129

Reply:
Ah now I think I get why you wanted to do this. Maybe better to just not include this field in the form. The same way we handle the `project` field (Take a look at the `ProjectEditForm`).